vertx-template-engines/vertx-web-templ-jte/src/main/java/io/vertx/ext/web/templ/jte/impl/VertxDirectoryCodeResolver.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2014 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.ext.web.templ.jte.impl;
+
+import gg.jte.CodeResolver;
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.common.WebEnvironment;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * @author <a href="mailto:andy@mazebert.com">Andreas Hager</a>
+ */
+public class VertxDirectoryCodeResolver implements CodeResolver {
+  private final Vertx vertx;
+  private final String templateRootDirectory;
+  private final ConcurrentMap<String, Long> modificationTimes;
+
+  public VertxDirectoryCodeResolver(Vertx vertx, String templateRootDirectory) {
+    this.vertx = vertx;
+    this.templateRootDirectory = templateRootDirectory.endsWith("/") ? templateRootDirectory : templateRootDirectory + "/";
```

Comment:
I think we should replace the `/` with the system property and this may not work on `Windows`

Reply:
I created the PR on a windows machine, seemed to work fine. I think Windows handles both \ or /.

Reply:
@pmlopes I've adjusted it to use the `Path` API instead of manual String concatenation. What do you think?
https://github.com/vert-x3/vertx-web/pull/1861/commits/ca3f7dbecaeeb62ac234a11b462855a11327e52c